src/graph/Graph.tsx

```diff
@@ -140,26 +146,37 @@ const GraphComponent = ({ data, config: cfg }: Props) => {
   }
 
   const onHighlightChange = (value: Highlight) => {
-    setGraph(resetHighlight)
+    setGraph(mergedGraph)
     setHighlight(value)
   }
 
+  const getHighlightEdges = (id: string) => {
```

Comment:
You may want to have less stuff inside of a component by decoupling it from state and moving it out somewhere else, maybe even at the end of a file

`getHighlightsEdges = (id: string, graph: Graph) => { ... }`